### PR TITLE
fix(max): Prevent certain events from confusing LLM

### PR DESCRIPTION
**Pull request by Twixes**
## Problem

The LLM gets confused when it sees PostHog events like `$feature_flag_called`, and often reasons they mean feature usage – whereas that's more of a PostHog implementation detail.

## Changes

Core filter definitions now can have the `ignored_in_assistant` flag. If they have it, the LLM is not fed this event's existence.

Also refactored `hardcoded_definitions.py` to `taxonomy.py`, since it was and is a carbon copy of `taxonomy.tsx` in the frontend.

### CODE REVIEW

File: ee/hogai/trends/nodes.py [hunk 1]
```diff
@@ -163,11 +163,12 @@ def _events_prompt(self) -> str:
 
         for event_name in events:
             event_tag = event_name
-            if event_name in hardcoded_prop_defs["events"]:
-                data = hardcoded_prop_defs["events"][event_name]
-                event_tag += f" - {data['label']}. {data['description']}"
-                if "examples" in data:
-                    event_tag += f" Examples: {data['examples']}."
+            if event_core_definition := CORE_FILTER_DEFINITIONS_BY_GROUP["events"][event_name]:
```
Comment (skoob13) [inline ee/hogai/trends/nodes.py:166]: Custom events will fail here.
```
if event_core_definition := CORE_FILTER_DEFINITIONS_BY_GROUP["events"].get(event_name):
```

File: ee/hogai/taxonomy.py [hunk 2]
```diff
@@ -28,29 +107,35 @@
...
             "description": "When a user opts into analytics.",
+            "ignored_in_assistant": True,  # Irrelevant product-wise
         },
         "$feature_flag_called": {
             "label": "Feature Flag Called",
             "description": (
                 'The feature flag that was called.\n\nWarning! This only works in combination with the $feature_flag event. If you want to filter other events, try "Active Feature Flags".'
             ),
             "examples": ["beta-feature"],
+            "ignored_in_assistant": True,  # Mostly irrelevant product-wise
```
Comment (skoob13) [inline ee/hogai/taxonomy.py:123]: I think we should keep this one. If we removed this event, we wouldn't be able to query the overall adoption of features. We can be more specific about this event by saying that this must be used only for queries where the overall adoption matters.

File: ee/hogai/taxonomy.py [hunk 3]
```diff
@@ -1025,3 +1111,38 @@
...
+            **value,
+            "label": f"Initial {value['label']}",
+            "description": (
+                f"{value['description']} Data from the first time this user was seen."
+                if "description" in value
+                else "Data from the first time this user was seen."
+            ),
+        }
+    else:
+        CORE_FILTER_DEFINITIONS_BY_GROUP["person_properties"][key] = value
```
Comment (skoob13) [inline ee/hogai/taxonomy.py:1137]: Is it not needed?

**Review by skoob13 — APPROVED** [review]
Looks great. There are couple minor comments.

File: ee/hogai/taxonomy.py (hunk 2 above)
Comment (Twixes) [reply ee/hogai/taxonomy.py:123]: Hmm, so I actually see the assistant using this for feature adoption as the problem I want to solve. $feature_flag_called is a really bad proxy for actual usage, as all it means is a feature flag was evaluated – not whether the feature was used. And it's only for flagged features too obviously.
We should get the assistant to determine activation events and actions instead, such as "insight created". And if no such metric is found, we should suggest instrumenting one

**Pull request merged**